ccip.go

```diff
@@ -0,0 +1,163 @@
+package ens
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/wealdtech/go-ens/v3/contracts/offchainresolver"
+	"github.com/wealdtech/go-ens/v3/contracts/universalresolver"
+)
+
+func getCcipReadError(err error) (bool, string) {
+	var jsonErr, ok = err.(rpc.DataError)
+	if !ok {
+		return false, ""
+	}
+	errData, ok := jsonErr.ErrorData().(string)
+	if !ok {
+		return false, ""
+	}
+	return (len(errData) >= 10 && errData[:10] == offchainLookupSignature), errData
+}
+
+func CCIPRead(backend bind.ContractBackend, rAddr common.Address, revertData string) ([]byte, error) {
+	hexBytes, err := hex.DecodeString(revertData[2:])
```

Comment:
Is this meant to be stripping `0x` from the beginning of the string?  If so then please use `strings.TrimPrefix()` as that protects against the `0x` not being present.